core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -2916,6 +2916,46 @@ public static long bitCount(ByteString b) {
     return bitsSet;
   }
 
+  /** Helper function for implementing MySQL <code>BIT_COUNT</code>. Counts the number
+   * of bits set in a boolean value. */
+  public static long bitCountMySQL(Boolean b) {
+    return Long.bitCount(b ? 1L : 0L);
+  }
+
+  /** Helper function for implementing MySQL <code>BIT_COUNT</code>. Counts the number
+   * of bits set in a string value. */
```

Comment:
if you can point to the MySQL documentation where this is specified, please add a link to the comments.

Reply:
This is MySQL BIT_COUNT function document - https://dev.mysql.com/doc/refman/8.4/en/bit-functions.html#bit-operations-bit-count, but unfortunately, it does not describe the details about the BIT_COUNT function. These special behaviors of BIT_COUNT come from actual tests with MySQL.